node-test/src/test/scala/org/bitcoins/node/NodeWithWalletTest.scala

```diff
@@ -58,8 +60,16 @@ class NodeWithWalletTest extends BitcoinSWalletTest {
 
         val onTx: DataMessageHandler.OnTxReceived = { tx =>
           if (expectedTxId.contains(tx.txId)) {
+            logger.debug(s"Cancelling timeout we set earlier")
             cancellable.map(_.cancel())
-            completionP.success(succeed)
+
+            for {
+              prevBalance <- wallet.getUnconfirmedBalance()
+              _ <- wallet.processTransaction(tx, confirmations = 0)
+              balance <- wallet.getUnconfirmedBalance()
+            } completionP.success(
```

Comment:
This seems like weird syntax, why no yield here?

Reply:
Also I can't get this test case to fail if I switch the predicate to be

```scala
assert(balance != prevBalance + amountFromBitcoind))
```

Reply:
Nice catch! You're right that this test was not falsifiable. This was because `assert` doesn't return anything if passed `false`, but throws a `TestFailedException`. The `.success(...)` method was therefore never evaluated. 

`completetionP.complete { Try { ... } } ` solved it!

Re. the syntax question: A `for` expression without a `yield` does a `foreach` instead on the final value instead of `map`. We don't use the end result for anything except for a side effect, so IMO not having a `yield` communicates intent better.